File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.py
class Solution(object):
    def minAddToMakeValid(self, s):
        _open, _close = 0, 0
        for c in s:
            if c == '(':
                _open += 1
            else:
                if _open > 0:
                    _open -= 1
                else:
                    _close += 1
        
        return _open + _close
        

File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.js
/**
 * @param {string} s
 * @return {number}
 */
var minAddToMakeValid = function(s) {
    let open = 0, close = 0;
    for(const c of s) {
        if(c == '(') open++;
        else {
            if(open > 0) open--;
            else close++;
        }
    }

    return open + close;
};